superset/assets/src/explore/components/controls/VizTypeControl.jsx

```diff
@@ -69,6 +69,53 @@ export default class VizTypeControl extends React.PureComponent {
       this.searchRef.focus();
     }
   }
+  buildVizTypeLookup(types) {
+    const lookup = new Map();
+    types.forEach((type) => {
+      lookup.set(type.key, type);
+    });
+
+    for (let i = 0; i < types.length; i++) {
```

Comment:
I don't think you need this second loop here

Reply:
The 2nd loop is to make sure any viz types that are not listed in the `defaultOrder` will be picked up.  Otherwise we will need to make sure the list actually contains all viz types and every time a new type is added, we also need to decide which position that viz type should be inserted into the `defaultOrder`.  We can probably add test to make sure that.  What do u think?

Reply:
That's right!